test/index.ts

```diff
@@ -11,25 +11,35 @@ describe('transforms', () => {
       return path.extname(file) === '.ts'
     })
     .forEach(file =>
-      (['ES5', 'ESNext'] as const).forEach(target =>
+      (['ES5', 'ESNext', 'Excluded'] as const).forEach(target =>
         it(`should transform ${file} as expected when target is ${target}`, async () => {
           let result = ''
           const fullFileName = path.join(fileTransformationDir, file),
             postCompileFullFileName = fullFileName.replace(/\.ts$/, '.js')
 
-          if ( path.basename(file) !== 'forOf-lengthCached.ts' ) {
-            compile(
-              [fullFileName],
-              ts.ScriptTarget[target],
-              (fileName, data) => postCompileFullFileName === path.join(fileName) && (result = data)
-            )
+            if (target !== 'Excluded') {
+              if ( path.basename(file) !== 'forOf-lengthCached.ts' ) {
+                compile(
+                  [fullFileName],
+                  ts.ScriptTarget[target],
+                  (fileName, data) => postCompileFullFileName === path.join(fileName) && (result = data)
+                )
+              } else {
+                // forOf-lengthCached
+                compileCached(
+                  [fullFileName],
+                  ts.ScriptTarget[target],
+                  (fileName, data) => postCompileFullFileName === path.join(fileName) && (result = data)
+                )
+              }
           } else {
-            // forOf-lengthCached
-            compileCached(
+            // check exclusions
+            compileExcluded(
               [fullFileName],
               ts.ScriptTarget[target],
               (fileName, data) => postCompileFullFileName === path.join(fileName) && (result = data)
             )
+            console.log(result)
```

Comment:
I added the exclusions option so we can exclude some of the transformations. However, I am not sure why for `Array.forOf` the result is still transformed

Reply:
I found the error. I had passed the wrong target.